### PR TITLE
Update semantic version from 1.0.0 to 2.0.0

### PR DESCRIPTION
**Pull request by agocke**
As it says

@ManishJayaswal @amcasey @tannergooding @tmeschter

### CODE REVIEW

**Comment by agocke** [comment]
retest this please

**Comment by agocke** [comment]
@dotnet-bot  retest this please

**Comment by ManishJayaswal** [comment]
@agocke  don't we need binding redirects for csc.exe.config?

File: src/Compilers/Core/VBCSCompiler/App.config [hunk 1]
```diff
@@ -5,6 +5,18 @@
   <runtime>
     <gcServer enabled="true"/>
     <gcConcurrent enabled="false"/>
+    <dependentAssembly>
+        <assemblyIdentity name="Microsoft.CodeAnalysis" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="1.0.0.0-1.0.65535.65535" newVersion="2.0.0.0" />
```
Comment (tmeschter) [inline src/Compilers/Core/VBCSCompiler/App.config]: @agocke Presumably we would want to redirect a binary built against any 1.* to be redirected. That is, shouldn't we be doing `oldVersion="1.0.0.0-1.65535.65535.65535"`?

Also, it looks like your formatting is a little off. Most of the file uses two spaces for indentation, not four.

**Comment by tmeschter** [comment]
@agocke Modulo my comment and @ManishJayaswal's comment about csc.exe.config, LGTM.

**Pull request merged**